tests/trainer/test_trainer.py

```diff
@@ -1177,81 +1177,75 @@ def val_dataloader(self):
+            dict(_strategy_type="single_device", _device_type=_AcceleratorType.GPU, num_gpus=1),
         ),
         (
             dict(accelerator="dp", gpus=1),
-            dict(_strategy_type=_StrategyType.DP, _device_type=_AcceleratorType.GPU, num_gpus=1, num_processes=1),
+            dict(_strategy_type="dp", _device_type=_AcceleratorType.GPU, num_gpus=1),
         ),
         (
             dict(accelerator="ddp", gpus=1),
-            dict(_strategy_type=_StrategyType.DDP, _device_type=_AcceleratorType.GPU, num_gpus=1, num_processes=1),
+            dict(_strategy_type="ddp", _device_type=_AcceleratorType.GPU, num_gpus=1),
         ),
         (
             dict(accelerator="ddp_cpu", num_processes=2, gpus=1),
-            dict(
-                _strategy_type=_StrategyType.DDP_SPAWN, _device_type=_AcceleratorType.CPU, num_gpus=0, num_processes=2
-            ),
+            dict(_strategy_type="ddp_spawn", _device_type=_AcceleratorType.CPU, num_gpus=0),
         ),
         (
             dict(accelerator="ddp2", gpus=1),
-            dict(_strategy_type=_StrategyType.DDP2, _device_type=_AcceleratorType.GPU, num_gpus=1, num_processes=1),
+            dict(_strategy_type="ddp2", _device_type=_AcceleratorType.GPU, num_gpus=1),
         ),
         (
             dict(accelerator=None, gpus=2),
-            dict(
-                _strategy_type=_StrategyType.DDP_SPAWN, _device_type=_AcceleratorType.GPU, num_gpus=2, num_processes=2
-            ),
+            dict(_strategy_type="ddp_spawn", _device_type=_AcceleratorType.GPU, num_gpus=2),
         ),
         (
             dict(accelerator="dp", gpus=2),
-            dict(_strategy_type=_StrategyType.DP, _device_type=_AcceleratorType.GPU, num_gpus=2, num_processes=1),
+            dict(_strategy_type="dp", _device_type=_AcceleratorType.GPU, num_gpus=2),
         ),
         (
             dict(accelerator="ddp", gpus=2),
-            dict(_strategy_type=_StrategyType.DDP, _device_type=_AcceleratorType.GPU, num_gpus=2, num_processes=2),
+            dict(_strategy_type="ddp", _device_type=_AcceleratorType.GPU, num_gpus=2),
         ),
         (
             dict(accelerator="ddp2", gpus=2),
-            dict(_strategy_type=_StrategyType.DDP2, _device_type=_AcceleratorType.GPU, num_gpus=2, num_processes=1),
+            dict(_strategy_type="ddp2", _device_type=_AcceleratorType.GPU, num_gpus=2),
         ),
         (
             dict(accelerator="ddp2", num_processes=2, gpus=None),
-            dict(_strategy_type=_StrategyType.DDP, _device_type=_AcceleratorType.CPU, num_gpus=0, num_processes=2),
+            dict(_strategy_type="ddp", _device_type=_AcceleratorType.CPU, num_gpus=0),
         ),
         (
             dict(accelerator="dp", num_processes=2, gpus=None),
-            dict(_strategy_type=_StrategyType.DDP, _device_type=_AcceleratorType.CPU, num_gpus=0, num_processes=2),
+            dict(_strategy_type="ddp", _device_type=_AcceleratorType.CPU, num_gpus=0),
```

Comment:
Why are we removing the `num_processes` param here?

Reply:
Because of this https://github.com/PyTorchLightning/pytorch-lightning/issues/11624
The current trainer.num_processes is incorrect and inconsistent, so I didn't try to reproduce it as we will deprecated it anyway